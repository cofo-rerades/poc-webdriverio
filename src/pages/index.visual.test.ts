import { $, browser, expect } from "@wdio/globals";

describe("Index page Visual regression", () => {
  it("should visual check  for a page ", async () => {
    await browser.url("/");
    const elem = await $("h1");
    expect(await elem.getText()).toBe("Search Flights");
    // take a visual snapshot of the whole screen
    await expect(browser).toMatchScreenSnapshot("index-page");
  });
  it("should visual check for an  individual element", async () => {
    await browser.url("/");
    // take a visual snapshot of an element
    await expect($("#departure")).toMatchElementSnapshot("departure-calendar");
  });

  it("should visual check for a state (hover) on an element", async () => {
    await browser.url("/");
    const elem = await $(".btn");
    elem.moveTo();
    // take a visual snapshot of an element
    await expect($(".btn")).toMatchElementSnapshot("hover-btn");
  });
});
