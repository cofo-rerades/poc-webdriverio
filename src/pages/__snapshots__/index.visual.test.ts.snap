// Snapshot v1

exports[`Index page Visual regression > should show index page  > index-page 1`] = `
Browser {
  "addCommand": [Function],
  "addLocatorStrategy": [Function],
  "capabilities": {
    "acceptInsecureCerts": true,
    "browserName": "chrome",
    "browserVersion": "134.0.6998.90",
    "chrome": {
      "chromedriverVersion": "134.0.6998.90 (219ee74681fc640b1ee1af1c99df3729a0ec25a8-refs/branch-heads/6998@{#1900})",
      "userDataDir": "/var/folders/25/h2z5y8yd5h13qz5fpv84cd4w0000gn/T/.org.chromium.Chromium.l933ca",
    },
    "fedcm:accounts": true,
    "goog:chromeOptions": {
      "debuggerAddress": "localhost:58405",
    },
    "networkConnectionEnabled": false,
    "pageLoadStrategy": "normal",
    "platformName": "mac",
    "proxy": {},
    "setWindowRect": true,
    "strictFileInteractability": false,
    "timeouts": {
      "implicit": 0,
      "pageLoad": 300000,
      "script": 30000,
    },
    "unhandledPromptBehavior": "dismiss and notify",
    "wdio:driverPID": 59032,
    "webauthn:extension:credBlob": true,
    "webauthn:extension:largeBlob": true,
    "webauthn:extension:minPinLength": true,
    "webauthn:extension:prf": true,
    "webauthn:virtualAuthenticators": true,
  },
  "overwriteCommand": [Function],
  "sessionId": "44fcecf2fce23d0e75d1fd9ed6f2b0f5",
}
`;
